content/en/docs/concepts/services-networking/gateway.md

```diff
@@ -0,0 +1,186 @@
+class. The GatewayClass can also define optional configuration parameters.
+
+A minimal GatewayClass example:
+
+{{% code_sample file="service/networking/gateway-class.yaml" %}}
+
+In this example, a controller that implements the Gateway API is configured to manage GatewayClasses
+with the controller name `example.com/gateway-controller`. An optional `parametersRef` is specified to
+reference an implementation-specific Parameters custom resource containing additional configuration
+parameters. Gateways of this class will be configured with these parameters.
+
+See the [GatewayClass][gatewayclass] reference for a full definition of this API kind.
+
+### Gateway {#api-kind-gateway}
+
+{{< gateway-feature-state for_gateway_version="v1.0" state="stable" >}}
+
+A Gateway describes an instance of traffic handling infrastructure. It defines a network endpoint
+that can be used for processing traffic, i.e. filtering, balancing, splitting, etc. for backends
+such as a Service resource. For example, HTTP traffic sent to a Kubernetes Service by a cloud load
+balancer or an in-cluster proxy server.
+
+A minimal Gateway resource example:
+
+{{% code_sample file="service/networking/gateway.yaml" %}}
+
+In this example, an instance of traffic handling infrastructure is programmed to listen for HTTP
+traffic on port 80. Since the `addresses` field is unspecified, an address or hostname is assigned
+to the Gateway by the controller. This address is used as a network endpoint for processing traffic
+of backend network endpoints.
+
+See the [Gateway][gateway] reference for a full definition of this API kind.
+
+### HTTPRoute {#api-kind-httproute}
+
+{{< gateway-feature-state for_gateway_version="v1.0" state="stable" >}}
+
+The HTTPRoute kind specifies routing behavior of HTTP requests from a Gateway listener to backend network
+endpoints. For a Service backend, an implementation may represent the backend network endpoint as a Service
+IP or the backing Endpoints of the Service. An HTTPRoute represents configuration that is applied to the
+underlying Gateway implementation. For example, defining a new HTTPRoute may result in configuring additional
+traffic routes into a cloud load balancer or in-cluster proxy server.
+
+A minimal HTTPRoute example:
+
+{{% code_sample file="service/networking/httproute.yaml" %}}
+
+In this example, HTTP traffic from Gateway `example-gateway` with the Host: header set to `www.example.com`
+and the request path specified as `/login` will be routed to Service `example-svc` on port `8080`.
+
+See the [HTTPRoute][httproute] reference for a full definition of this API kind.
+
+## Request flow
+
+Here is a simple example of HTTP traffic being routed to a Service by using a Gateway and an HTTPRoute:
+
+{{< figure src="/docs/images/gateway-request-flow.svg" alt="gateway-request-flow" class="diagram-medium" link="https://raw.githubusercontent.com/kubernetes/website/main/content/en/docs/images/gateway-request-flow.svg" >}}
```

Comment:
- Don't link to raw GitHub content like this; people might bookmark it and we don't want that.
- Please set alt attributes that are meaningful for people with accessibility needs